managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/EntryCacheManager.java

```diff
@@ -45,9 +45,9 @@
 @SuppressWarnings("checkstyle:javadoctype")
 public class EntryCacheManager {
 
-    private final long maxSize;
-    private final long evictionTriggerThreshold;
-    private final double cacheEvictionWatermark;
+    private volatile long maxSize;
+    private volatile long evictionTriggerThreshold;
+    private volatile double cacheEvictionWatermark;
```

Comment:
@merlimat - do you have any thoughts here? I remember that you have raised concerns about `volatile` in the metadatastore.

I know that we are talking about minor optimizations here, but since this is on the write path, I feel it makes sense to discuss now.

Reply:
> do you have a use case that shows why the benefits out way the costs here?

If we need to adjust the cache parameters, we need to restart the online cluster in turn. Restarting the cluster will have a certain impact on online business.

Reply:
> I am concerned about the cost of this change. These three variables are read each time that a message is written to the cache. While I see the generic benefit of increasing configurability, I don't think we should do so in a way that adds cost (even small costs) to the write path.

Do you have some good suggestions? About meeting the needs of dynamic configuration cache, but also solving the problem you said.